File: internal/adr/adr.go
package adr

import (
	"github.com/delightfulhammers/telesis/internal/docgen"
)

var cfg = docgen.Config{
	Prefix:   "ADR",
	Subdir:   "adr",
	Template: "adr.md.tmpl",
}

// Create creates a new ADR file with the next sequential number.
// It returns the path to the created file.
func Create(rootDir, slug string) (string, error) {
	return docgen.Create(rootDir, cfg, slug)
}

// NextNumber scans the ADR directory and returns the next sequential number.
func NextNumber(adrDir string) (int, error) {
	return docgen.NextNumber(adrDir, cfg.Prefix)
}

File: src/adr/adr.ts
import { create as docCreate, nextNumber as docNextNumber } from "../docgen/docgen.js";
import type { DocConfig } from "../docgen/docgen.js";

const ADR_CONFIG: DocConfig = {
  prefix: "ADR",
  subdir: "adr",
  template: "adr.md.tmpl",
};

export const create = (rootDir: string, slug: string): string =>
  docCreate(rootDir, ADR_CONFIG, slug);

export const nextNumber = (adrDir: string): number =>
  docNextNumber(adrDir, "ADR");
